src/coreclr/jit/emit.cpp

```diff
@@ -4102,6 +4199,382 @@ void emitter::emitDispCommentForHandle(size_t handle, GenTreeFlags flag)
+#endif
+
+                        // remove the instruction from the group
+                        group->igInsCnt -= 1;
+                        group->igSize -= (unsigned short)sizeRemoved;
+                        emitTotalCodeSize -= sizeRemoved;
+
+                        // flag the group as resized and recounted
+                        group->igFlags |= (IGF_UPD_ISZ | IGF_UPD_ICOUNT);
+
+                        // cleanup
+                        instructionDescriptor = nullptr;
+                        removedCount += 1;
+                    }
+#if DEBUG
+                    else if (EMITVERBOSE)
+                    {
+                        printf("IN%04x is not the last instruction in IG%02u, adding to jump list.\n",
+                               jmp->idDebugOnlyInfo()->idNum, jmp->idjIG->igNum);
+                    }
+#endif
+                }
+
+#if DEBUG
+                else if (EMITVERBOSE)
+                {
+                    printf("IN%04x is the only instruction in IG%02u, adding to jump list.\n",
+                           jmp->idDebugOnlyInfo()->idNum, jmp->idjIG->igNum);
+                }
+#endif
+            }
+#if DEBUG
+            else if (EMITVERBOSE)
+            {
+                if (targetGroup == nullptr)
+                {
+                    printf("IN%04x from the last instruction in IG%02u target is not set!, adding to jump list.\n",
+                           jmp->idDebugOnlyInfo()->idNum, jmp->idjIG->igNum);
+                }
+                else if ((jmp->idjIG->igFlags & IGF_HAS_ALIGN) != 0)
+                {
+                    printf("IN%04x from the last instruction in IG%02u contains alignment, adding to jump list.\n",
+                           jmp->idDebugOnlyInfo()->idNum, jmp->idjIG->igNum);
+                }
+                else if (jmp->idjIG->igNext != targetGroup)
+                {
+                    printf("IN%04x from the last instruction in IG%02u to does not jump to the next group, adding to "
+                           "jump list.\n",
+                           jmp->idDebugOnlyInfo()->idNum, jmp->idjIG->igNum);
+                }
+            }
+#endif
+        }
+
+        if (jmp != nullptr)
+        {
+            lastInserted = emitInsertJumpAlwaysIntoJumpList(jmp, lastInserted);
+#ifdef DEBUG
+            {
+                bool found = false;
+                for (instrDescJmp* cur = emitJumpList; cur != nullptr; cur = cur->idjNext)
+                {
+                    if (cur == jmp)
+                    {
+                        found = true;
+                        break;
+                    }
+                }
+                if (!found)
+                {
+                    assert(!"jmp failed to add to emitJumpList");
+                }
+            }
+#endif
+        }
+    }
+    if (previousTruncatedGroup != nullptr)
+    {
+        for (insGroup* offsetGroup = previousTruncatedGroup->igNext; offsetGroup != nullptr;
```

Comment:
Add a comment that this adjusts the offset of remaining IGs until the end.

Reply:
Done